validator/client/propose.go

```diff
@@ -81,6 +97,15 @@ func (v *validator) ProposeBlock(ctx context.Context, slot types.Slot, pubKey [4
 		return
 	}
 
+	// processPandoraShardHeader method process the block header from pandora chain
+	if status, err := v.processPandoraShardHeader(ctx, b, slot, epoch, pubKey); !status || err != nil {
```

Comment:
I am not 100% sure if we should use the word `pandora` in our function names and not simple: `v.processShardHeader`

as in the future other systems could create them too. Whats your ideas here @blazejkrzak @mxmar ?
Maybe it makes sense to keep it pandora specific, as for other shard types we will have a new name...

Reply:
processShardHeader, then if another being comes around use strategy pattern

Reply:
My guess, in future other sharding chain will not be verified as like as pandora chain in block propose functionality.